Source/src/components/ComponentProgressBar.cpp

```diff
@@ -0,0 +1,91 @@
+#include "core/hepch.h"
+#include "ComponentProgressBar.h"
+
+Hachiko::ComponentProgressBar::ComponentProgressBar(GameObject* container) : Component(Type::PROGRESS_BAR, container) 
+{
+	// Get backgroundID;
+	// Get foregroundID;
+	// Get background;
+	// Get foreground;
+}
+
+void Hachiko::ComponentProgressBar::Update() 
+{
+    if (game_object->children.size() < 2)
```

Comment:
I don't see how this validation would make this to actually work, the children can be something else, we should check for particular components on the structure